cart_robot:
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50
  # Position Controllers ---------------------------------------
  up_left_wheel_controller:
    type: effort_controllers/JointVelocityController
    joint: wheel_up_left_joint
    pid: {p: 5.0, i: 20, d: 0.0, i_clamp: 100.0}
  up_right_wheel_controller:
    type: effort_controllers/JointVelocityController
    joint: wheel_up_right_joint
    pid: {p: 5.0, i: 20, d: 0.0, i_clamp: 100.0}

  down_left_wheel_controller:
    type: effort_controllers/JointVelocityController
    joint: wheel_down_left_joint
    pid: {p: 5.0, i: 20, d: 0.0, i_clamp: 100.0}
  down_right_wheel_controller:
    type: effort_controllers/JointVelocityController
    joint: wheel_down_right_joint
    pid: {p: 5.0, i: 20, d: 0.0, i_clamp: 100.0}
  
  DifferentialDriveController:
    type: "diff_drive_controller/DiffDriveController"
    left_wheel_joint: wheel_up_left_joint
    right_wheel_joint: wheel_up_right_joint
    publish_rate: 50
    # wheel_separation: 0.4
    # cmd_vel_topic: /cmd_vel
    odom_topic: /odom
    odom_frame: odom
    robot_base_frame: base_link
    enable_odom_tf: true
    imu_topic: /imu
    imu_frame: imu_link
    enable_imu_tf: true
    wheel_separation: 0.25  # Update this value to match your robot's wheel separation
    cmd_vel_topic: /cmd_vel  # Subscribe to the /cmd_vel topic
